### PR TITLE
Re-implement RRD charts in Chart.js

### LINKED ISSUE #2859: [Feature] Improve graphs on "Task Monitor" page
### Describe the feature or change in detail

The graphs on the "Task Monitor" page are basic static images that get reloaded with the page every 10 seconds or so:

image

It would be great if we could get real graphs, even better if the data was somewhat realtime. WordOps for example has interactive graphs ([demo](https://demo.wordops.eu/)).

I'm not sure how accessible this data is, if it can be streamed to the page etc. but I thought I'd put in this feature request as I think it would improve the Hestia experience.

### Would you like to sponsor this feature to have it implemented?

No

### PR DESCRIPTION
**Pull request by AlecRust**
Re-implements the RRD charts as Chart.js v4 charts using the foundation code provided by @jaapmarcus.

I've tried to keep the charts looking like the old RRD images as much as possible. Any testing/feedback appreciated.

**Before:**
image

**After:**
image

---

closes #2859

### CODE REVIEW

File: web/js/pages/list_rrd.js [hunk 1]
```diff
@@ -57,11 +57,40 @@ function formatLabel(date, period) {
 }
 
 function getChartOptions() {
+	const currentTheme = getCurrentTheme();
+	const textColor = currentTheme === 'dark' ? '#cdcdcd' : '#7c7c7c';
```
Comment (jaapmarcus) [inline web/js/pages/list_rrd.js]: If users create his own dark theme this check will probably not work...

How ever I don' t know if there is a easy fix for it..

Comment (AlecRust) [reply web/js/pages/list_rrd.js]: Good point. I've refactored this to use CSS variables instead, which can be more easily overridden.

**Review by jaapmarcus — APPROVED** [review]

File: web/list/rrd/ajax.php [hunk 2]
```diff
@@ -9,28 +9,42 @@
...
 	$output,
 	$return_var,
 );
+
+if ($return_var != 0) {
+	http_response_code(500);
+	exit("Error fetching RRD data");
+}
+
+$serviceUnits = [
```
Comment (AlecRust) [inline web/list/rrd/ajax.php:40]: I guess this is ok..? I wanted to keep the Y axis labels which weren't in the data. Not sure I've covered all bases here.

File: web/list/rrd/ajax.php [hunk 3]
```diff
@@ -9,28 +9,42 @@
...
+	"la" => "Points",
+	"mem" => "Mbytes",
+	"net_enp0s1" => "KBytes",
```
Comment (jaapmarcus) [inline web/list/rrd/ajax.php]: Changes on the network port the user as it makes more sense to match it against net_ instead define each possible network port name...

**Review by jaapmarcus — COMMENTED** [review]
See comment

File: web/list/rrd/ajax.php (hunk 3 above)
Comment (jaapmarcus) [reply web/list/rrd/ajax.php]: or net will probally also work

Comment (AlecRust) [reply web/list/rrd/ajax.php]: Fixed I think

**Comment by jlguerrerosanz** [comment]
Beautiful work. Thank you
